File: BOJ/Python/2428-표절.py
n = int(input())
sizes = [int(i) for i in input().split()]
sizes.sort()
def lowerBound(target, left, right):
    while left < right:
        mid = (left + right)//2
        if sizes[mid] >= target:
            right = mid
        else:
            left = mid + 1
    return right
cnt = 0
for i in range(n):
    a = lowerBound(sizes[i]*0.9, 0, i)
    cnt += i-a
print(cnt)

File: BOJ/C++/2428-표절.cpp
#include <cstdio>
#include <algorithm>

int main() {
    int n, sizes[100001], tmp;
    scanf("%d", &n);
    for(int i = 0; i < n; i++) {
        scanf("%d", &tmp);
        sizes[i] = tmp;
    }
    std::sort(sizes, sizes+n);
    int cnt = 0, *a, res;
    float tmp2;
    for(int i = 0; i < n; i++) {
        tmp2 = (float)sizes[i]*0.9;
        a = std::lower_bound(sizes, sizes + n, tmp2);
        res = a - sizes + 1;
        cnt += i - res + 1;
    }
    printf("%d", cnt);
    return 0;
}